benchmarks/modules/utils.py

```diff
@@ -267,11 +268,25 @@ def setup_spack_environment(self):
             f'(cd {cp_dir}; find . \( -name "spack.yaml" -o -name "compilers.yaml" -o -name "packages.yaml" \) -print0 | xargs -0 tar cf - | tar -C {dest} -xvf -)',
             f'spack -e {self.build_system.environment} config add "config:install_tree:root:{env_dir}/opt"',
         ]
-
+        cmd_spack_spec_dict =    'from spack import environment;\
+                                spec_list = environment.active_environment().concretized_user_specs;\
```

Comment:
We debugged this together offline, I think this should be [`environment.active_environment().concrete_roots()`](https://spack.readthedocs.io/en/latest/spack.environment.html#spack.environment.Environment.concrete_roots):
```suggestion
                                spec_list = environment.active_environment().concrete_roots();\
```